src/modules/LanItems.rb

```diff
@@ -1232,95 +1226,35 @@ def Import(settings)
       true
     end
 
-    def GetDescr
-      descr = []
-      Builtins.foreach(
-        Convert.convert(
-          @Items,
-          from: "map <integer, any>",
-          to:   "map <integer, map <string, any>>"
-        )
-      ) do |key, value|
-        if Builtins.haskey(value, "table_descr") &&
-            Ops.greater_than(
-              Builtins.size(Ops.get_map(@Items, [key, "table_descr"], {})),
-              1
-            )
-          descr = Builtins.add(
-            descr,
-            "id"          => key,
-            "rich_descr"  => Ops.get_string(
-              @Items,
-              [key, "table_descr", "rich_descr"],
-              ""
-            ),
-            "table_descr" => Ops.get_list(
-              @Items,
-              [key, "table_descr", "table_descr"],
-              []
-            )
-          )
-        end
+    def GetDescr(overviews)
+      overviews.map do |key, value|
+        {
+          "id"          => key,
+          "rich_descr"  => value["table_descr"]["rich_descr"] || "",
+          "table_descr" => value["table_descr"]["table_descr"] || []
+        }
       end
-      deep_copy(descr)
     end
 
-    def needFirmwareCurrentItem
-      need = false
-      if IsNotEmpty(Ops.get_string(@Items, [@current, "hwinfo", "driver"], ""))
-        if Builtins.haskey(
-          @request_firmware,
-          Ops.get_string(@Items, [@current, "hwinfo", "driver"], "")
-        )
-          need = true
-        end
+    # Checks whether device need (known) firmware to be loaded
+    def needFirmwareCurrentItem(iface)
+      need = if iface.hardware.driver
```

Comment:
can you rename variable to something like driver?

Reply:
ok